src/compiler/types.ts

```diff
@@ -1861,6 +1861,53 @@ namespace ts {
         getJsxIntrinsicTagNames(): Symbol[];
         isOptionalParameter(node: ParameterDeclaration): boolean;
 
+        /**
+         * Two types are identical iff they are references to the exact same fully qualified, resolved type (usually an alias to ===)
+         */
+        isIdenticalTo(a: Type, b: Type): boolean;
+        /**
+         * A type is a subtype of another if the collection of its members is contained within the collection of the other's members
+         */
+        isSubtypeOf(a: Type, b: Type): boolean;
+        /**
+         * Same as subtype, but all types are assignable to `any` and `any` is assignable to all types
+         */
+        isAssignableTo(a: Type, b: Type): boolean;
+        /**
+         * Same as assignable, but all unions with overlapping constituient types are comparable, and units of the same type are comparable

```

Comment:
@DanielRosenwasser Listing examples where each relationship is used in the language would be a great thing to place into the documentation for each. Unfortunately, they way our spec is worded (and not up to date with the comparable changes), I'm not sure what is used where at present. If you have insights here, I'd be happy to add them.